### ISSUE TITLE
Remaining Ruby 2.7 deprecation warning

### ISSUE THREAD

**Issue by swrobel** [teamcapybara/capybara#2314]
## Meta
Capybara Version: 3.31.0
Ruby Version: ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-darwin18]

```
capybara-3.31.0/lib/capybara/session.rb:753: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
```

**Comment by twalpole** [comment]
That line is already using the Ruby 2.7 `(...)` syntax to forward all parameters - https://github.com/teamcapybara/capybara/blob/master/lib/capybara/session.rb#L753 - so the warning is likely to be from wherever it's being called which I can't tell from the limited info provided.

**Comment by swrobel** [comment]
Hmm, the only other output from the same spec is `capybara-3.31.0/lib/capybara/node/finders.rb:247: warning: The called method `find_all' is defined here`. Not sure if that's relevant or an unrelated warning.

**Comment by twalpole** [comment]
Not really -- that's what's getting called -- so something somewhere is calling `all`/`find_all` and not passing keyword arguments with `**` - what is the spec doing?

**Comment by swrobel** [comment]
Here's the offending line:

```ruby
date = all('div', class: %w[DayPicker-Day !DayPicker-Day--disabled !DayPicker-Day--outside]).sample
```

**Comment by swrobel** [comment]
Changing to this doesn't seem to prevent the warning:

```ruby
date = all('div', **{ class: %w[DayPicker-Day !DayPicker-Day--disabled !DayPicker-Day--outside] }).sample
```

**Comment by twalpole** [comment]
I can't replicate with that line - so maybe it's not coming from that line -- can you a `puts 'xxxx'` before and after that line and see if the warning is actually from that line

**Comment by swrobel** [comment]
Exactly what I did:

```
>>>>>>>>>>>>> finding
capybara-3.31.0/lib/capybara/session.rb:753: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
capybara-3.31.0/lib/capybara/node/finders.rb:247: warning: The called method `find_all' is defined here
<<<<<<<<<<<<<<<<<< found
```

**Comment by swrobel** [comment]
Sorry this is turning into a wild goose chase. Also added some puts to the gem code in session.rb just to be sure the 2.7 detection was working:

```
puts RUBY_VERSION
puts RUBY_VERSION >= '2.7'

2.7
true
```

**Comment by twalpole** [comment]
hmmm -- I can't reproduce with a simple use case - can you just step through the call and see when the error is generated?

**Comment by twalpole** [comment]
Looking through the code it's possible it's the matcher proxies - https://github.com/teamcapybara/capybara/blob/master/lib/capybara/rspec/matcher_proxies.rb - since they didn't get updated.  Later I'll try writing up a more complicated test to verify.

**Comment by swrobel** [comment]
Here's where it actually occurred, although I saw the `matcher_proxies.rb` file come up along the way and had the same thought...

```
[740, 749] in capybara-3.31.0/lib/capybara/session.rb
   740:     def save_and_open_screenshot(path = nil, **options)
   741:       save_screenshot(path, **options).tap { |s_path| open_file(s_path) } # rubocop:disable Lint/Debugger
   742:     end
   743:
   744:     def document
=> 745:       @document ||= Capybara::Node::Document.new(self, driver)
   746:     end
   747:
   748:     NODE_METHODS.each do |method|
   749:       if RUBY_VERSION >= '2.7'
(byebug) step
capybara-3.31.0/lib/capybara/session.rb:753: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
capybara-3.31.0/lib/capybara/node/finders.rb:247: warning: The called method `find_all' is defined here

[243, 252] in capybara-3.31.0/lib/capybara/node/finders.rb
   243:       #   @yieldparam element [Capybara::Node::Element]  The element being considered for inclusion in the results
   244:       #   @yieldreturn [Boolean]                     Should the element be considered in the results?
   245:       # @return [Capybara::Result]                   A collection of found elements
   246:       # @raise [Capybara::ExpectationNotMet]         The number of elements found doesn't match the specified conditions
   247:       def all(*args, allow_reload: false, **options, &optional_filter_block)
=> 248:         minimum_specified = options_include_minimum?(options)
   249:         options = { minimum: 1 }.merge(options) unless minimum_specified
   250:         options[:session_options] = session_options
   251:         query = Capybara::Queries::SelectorQuery.new(*args, **options, &optional_filter_block)
   252:         result = nil
```

**Comment by twalpole** [comment]
I've reproduced it, it's definitely the matcher proxies -- I'll fix it within a couple of days -- thanks for helping track it down

**Comment by twalpole** [comment]
@swrobel Please try the `issue_2314` and see if it fixes the issue

**Comment by swrobel** [comment]
@twalpole yes, thank you!

**Comment by twalpole** [comment]
Fixed via PR #2315
